File: ruby/leetcode/maximum-average-subarray-i.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Float}
def find_max_average(nums, k)
    this_sum = 0.0
    for i in 0...k do
        this_sum += nums[i]
    end
    max_sum = this_sum
    for i in k...nums.size() do
        this_sum += nums[i] - nums[i - k]
        max_sum = [max_sum, this_sum].max
    end
    return max_sum / k
end

File: python/leetcode/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        this_sum = 0.0
        for i in range(k):
            this_sum += nums[i]
        max_sum = this_sum
        for i in range(k, len(nums)):
            this_sum += nums[i] - nums[i - k]
            max_sum = max(max_sum, this_sum)
        return max_sum / k
